month_end/models/month_end.py

```diff
@@ -0,0 +1,125 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2017 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Maxime Beck
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+import datetime
+import calendar
+from odoo import api, models, fields
+
+
+class MonthEnd(models.Model):
+    _name = 'month.end'
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.model
+    def process_create_accounting_entry(self):
+        date_format = '%d.%m.%Y'
+
+        today = datetime.date.today()
+        first_day = today.replace(day=1)
+        last_day = today.replace(day=calendar.monthrange(today.year, today.month)[1])
+
+        first_day_last_month = first_day.replace(month=today.month-1)
+        last_day_last_month = last_day.replace(month=today.month-1)
+
+        analytic_account_cd_pgm = self.env['account.analytic.account'].search([('name', '=', 'CD pgm')])
+        partner_compassion_internationnal = self.env['res.partner'].search([('name', '=', 'Compassion Internationnal')])
+        
+        ## account_move_line entry
+        # gift entry
+        gift_account = self.env['account.account'].search([('code', '=', '2002')])
+        montly_gifts = self.env['sponsorship.gift'].search(
+            [
+                ('state', '=', 'In Progress'),
+                ('create_date', '>=', fields.Date.to_string(first_day)),
+                ('create_date', '<=', fields.Date.to_string(last_day))
+            ]
+        )
+        gift_name = 'gift invoice: ' + str(len(montly_gifts)) + ' gifts from ' + first_day.strftime(date_format) + ' to ' + last_day.strftime(date_format)
+        gift_debit = 0
+        for gift in montly_gifts:
+            gift_debit += gift.amount
+        gift_line_vals = {
+            'account_id': gift_account.id,
+            'name': gift_name,
+            'debit': gift_debit,
+            'partner_id': partner_compassion_internationnal.id
+        }
+
+        # sponsorship entry
+        sponsorship_account = self.env['account.account'].search([('code', '=', '5000')])
+        sponsorship_active = self.env['recurring.contract'].search_count([('child_id', '!=', False)])
+        sponsorship_disbursed = self.env['recurring.contract'].search_count([('child_id.project_id.hold_cdsp_funds', '=', False)])
+        sponsorship_name = str(sponsorship_active) + ' sponsorships, ' + \
+                           str(sponsorship_disbursed) + ' disbursed'
+
+        sponsorships = self.env['recurring.contract'].search([])
+        sponsorship_debit = 0
```

Comment:
No, you are putting all sponsorships in the debit value! I would rather do `sponsorship_debit = sum(sponsorship_disbursed.mapped('contract_line_ids').filtered(lambda line: line.product_id.property_account_expense_id == sponsorship_account.id).mapped('subtotal'))` 😆